src/utils/jwt-utils.ts

```diff
@@ -0,0 +1,52 @@
+import crypto from "crypto"
+
+import jwt from "jsonwebtoken"
+import _ from "lodash"
+
+import { config } from "@config/config"
+
+const JWT_SECRET = config.get("auth.jwtSecret")
+const ENCRYPTION_SECRET = config.get("auth.encryptionSecret")
+const AUTH_TOKEN_EXPIRY_MS = config.get("auth.tokenExpiryInMs").toString()
+
+export default {
+  decodeToken: _.wrap(jwt.decode, (decode, token: string) => decode(token)),
+  verifyToken: _.wrap(jwt.verify, (verify, token: string) =>
+    verify(token, JWT_SECRET, { algorithms: ["HS256"] })
+  ),
+  signToken: _.wrap(jwt.sign, (sign, content: string) =>
+    sign(content, JWT_SECRET, {
+      algorithm: "HS256",
+      expiresIn: AUTH_TOKEN_EXPIRY_MS,
+    })
+  ),
+  encryptToken: (token: string, secret?: string) => {
+    const encryptionSecret = secret ?? ENCRYPTION_SECRET
+
+    const cipher = crypto.createCipher(
```

Comment:
`crypto.createCipher` is deperciated. lets use `createCipheriv()` instead ya

Reply:
we can't, see PR description

> we are not using gcm or using an iv because this incurs downtime on our app (the case where you have old version of BE required to decrypt smth encrypted w/ iv or gcm will never work hence requiring downtime)

i've double-checked w @spaceraccoon on this and we're aligned that this is ok - the threat vector we want to mitigate is leaking plaintext creds, which this does help with. 

sadly using `iv` and/or `gcm` will always incur downtime so this was a conscious choice.

Reply:
@seaerchin 
no bah this just changing the function, iv can be null when calling `createCipheriv()` according to the [docs](https://nodejs.org/docs/latest-v18.x/api/crypto.html#cryptocreatecipherivalgorithm-key-iv-options) 

> If the cipher does not need an initialization vector, iv may be null.